ssl/statem/statem_quic.c

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "../ssl_locl.h"
+#include "statem_locl.h"
+#include "internal/cryptlib.h"
+
+#ifdef OPENSSL_NO_QUIC
+NON_EMPTY_TRANSLATION_UNIT
+#else
+
+/*
+ * The message should already be in the init_buf, we just
+ * have to deal with hashing, etc
```

Comment:
Bad comment (when copied from original source)

Reply:
Does some of this come directly out of BoringSSL? Is there going to be a licence issue? Although I think we're ok if a Google person says its ok.

Reply:
We have previous statements from David Benjamin and Adam Langley, that we can use their code with APL.

Reply:
Looking at it again, it was a comment based on what my intended implementation of the function was going to be (i.e. put data into `init_buf` like BoringSSL put the data into `s3->hs_buf`), but I swapped it around without removing the comment.

This is an OpenSSL-specific implementation of the BoringSSL QUIC APIs, only the most simple of APIs (e.g. `SSL_quic_write_level()`) are the same.